docs/devGuide/githubActions/workflowSecurity.md

```diff
@@ -0,0 +1,82 @@
+
+GitHub Actions are a powerful tool for automating workflows. However, it is important to ensure that your workflows are secure to prevent malicious actors from exploiting them.
+
+**Here are some best practices to follow when working on GitHub Actions workflows.**
+
+### Minimally scoped credentials
+Every credential used in the workflow should have the minimum required permissions to execute the job. Without explicit specification of permissions, the default permissions are used, which are usually more permissive than necessary.
+
+Use the ‘permissions’ key to make sure the `GITHUB_TOKEN` is configured with the least privileges for each job.
+
+<box type="tip" light>
+
+`permissions` can be restricted at the repo, workflow or job level. Pick the most restrictive level that still allows the workflow to function.
+
+</box>
+
+**Environment variables**, like {% raw %} `${{ secrets.GITHUB_TOKEN }}` {% endraw %}, should be limited by scope, and should be declared at the step level when possible.
+
+### Pull_request_target event
+Maximum access for pull requests from public forked repositories is read access due to security reasons. Since MarkBind uses the forking workflow, this leads to limitations in the CI/CD pipeline. 
+
+When a workflow is triggered by the `pull_request_target` event, the `GITHUB_TOKEN` is granted read/write repository permission, even when it is triggered from a public fork. However, this could lead to security vulnerabilities if not handled properly.
+
+This event should not be used with `actions/checkout` as it can give write permission and secrets access to untrusted code from the forked code. Any building step, script execution, or even action call could be used to compromise the entire repository.
+
+This can be fixed by adding code to ensure that the code being checked out belongs to the base branch.
+This can be done using:
+{% raw %}
+```yaml
+- uses: actions/checkout@v4
+  with:
+    ref: ${{ github.base_ref }}
+```
+<box type="warning">
+{% endraw %}
+
+This method could be limiting since the code checked out is not up to date for the pull request. This means that the code checked out is not the code that is being tested. This could lead to false positives or false negatives in the testing process.
+
+</box>
+
+This method triggers workflows based on the latest commit of the pull request's base branch. Even if workflow files are modified or deleted on feature branches, workflows on the default branch aren't affected so you can prevent malicious code from being executed in <tooltip content="Continuous Integration">CI</tooltip> without code review.
+
+Another solution that allows `pull_request_target` to work securely with `actions/checkout` used on the pull request branch, is to add an additional step of running workflow only on approval by trusted users, such that the trusted user has to check the changes in the code from the pull request to ensure there is no malicious code before running the workflow. 
```

Comment:
Perhaps paraphrasing this sentence into a few shorter sentences could make it more readable